src/3d/castletransform_physics.inc

```diff
@@ -501,17 +501,14 @@ end;
 
 procedure TRigidBody.DeinitializeTransform(const Transform: TCastleTransform);
 begin
+  if FRecreateKraftInstance then
```

Comment:
I'm uncertain of this line, I will need more time to investigate is it 100% correct. I mean, `DeinitializeTransform` is called from various occasions. 

- E.g. when you call `MyTransform.RigidBody := nil` in the middle of the game (when physics sumulation started).
- Or when you call `SceneManager.Items.Remove(MyTransform)` in the middle of the game.

I want to check (read the code & do the tests) what is the value of `FRecreateKraftInstance` in this case (and if it can be `true`, is it correct that we abort the rest of DeinitializeTransform work?)

Reply:
I added this check because in some cases deinitialisation was called twice.  I think `ChangeWorld` or `PhysicsChangeWorldDetach` need more checks to fix that. This need more investigation.

Reply:
Fixed in https://github.com/castle-engine/castle-engine/pull/118/commits/27df87cd0b18a824b111fe56843a71cf11738011 :)